lib/orderbook/TradingPair.ts

```diff
@@ -28,31 +28,36 @@ type OrderSidesQueues = {
   sell: FastPriorityQueue<StampedOrder>,
 };
 
-/** A class to represent a matching engine responsible for matching orders for a given trading pair according to their price and quantity. */
-class MatchingEngine {
+/**
+ * A class to represent a trading pair.
+ * responsible for managing all active orders, and for matching orders according to their price and quantity.
+ */
+class TradingPair {
   /** A pair of priority queues for the buy and sell sides of this trading pair */
-  public queues: OrderSidesQueues;
-  /** A pair of maps between active own orders ids and orders for the buy and sell sides of this trading pair. */
-  public ownOrders: OrderSidesLists<orders.StampedOwnOrder>;
-  /** A map between peerPubKey and a pair of maps between active peer orders ids and orders for the buy and sell sides of this trading pair. */
-  public peersOrders: Map<string, OrderSidesLists<orders.StampedPeerOrder>>;
-
-  constructor(private logger: Logger, public pairId: string) {
-    this.queues = {
-      buy: MatchingEngine.createPriorityQueue(OrderingDirection.Desc),
-      sell: MatchingEngine.createPriorityQueue(OrderingDirection.Asc),
-    };
+  public queues: OrderSidesQueues | undefined;
+  /** a pair of maps between active own orders ids and orders for the buy and sell sides of this trading pair. */
+  public ownOrders: OrderSidesMaps<orders.StampedOwnOrder>;
+  /** a map between peerPubKey and a pair of maps between active peer orders ids and orders for the buy and sell sides of this trading pair. */
+  public peersOrders: Map<string, OrderSidesMaps<StampedPeerOrder>>;
+
+  constructor(private logger: Logger, public pairId: string, private matching = true) {
+    if (matching) {
```

Comment:
I think I'd prefer sticking with `nomatching`, I know it raises the possibility for double negatives but I think it's better to call it the same thing everywhere, also this way we could actually get rid of the double negatives when passing in `!this.nomatching` to the constructor. It's fine if you prefer it this way though.